packages/grid_client/src/helpers/utils.ts

```diff
@@ -63,6 +63,25 @@ function zeroPadding(length: number, num: number) {
   return num.toString(16).padStart(length, "0");
 }
 
+/**
+ * Converts an object into a query string format.
+ * @param obj - The object to be converted into a query string.
+ * @returns A string representing the object in query string format.
+ */
+function convertObjectToQueryString(obj: Record<string, any>): string {
+  // Filter out undefined or empty string values
+  const filteredParams = Object.fromEntries(
+    Object.entries(obj).filter(([_, value]) => value !== undefined && value !== ""),
+  );
+
+  // Convert object to query string format
+  const queryString = Object.entries(filteredParams)
+    .map(([key, value]) => `${encodeURIComponent(key)}=${encodeURIComponent(value)}`)
```

Comment:
Why do we use encodeURIComponent?

Reply:
Its a built in helper used to encode special characters in a URL.

e.g.

```ts
const searchTerm = "hello world!";
const url = "https://example.com/search?q=" + encodeURIComponent(searchTerm);
console.log(url);
// Output: https://example.com/search?q=hello%20world%21
```